lib/sidekiq/history/statistic/runtime_statistic.rb

```diff
@@ -18,38 +18,34 @@ def values_hash
       end
 
       def max_runtime
-        values.max || 0.0
+        values(:max_time).max || 0.0
       end
 
       def min_runtime
-        values.min || 0.0
+        values(:min_time).min || 0.0
       end
 
       def last_runtime
         @redis_statistic
-          .for_worker(@worker)
-          .map{ |s| s[:last_runtime] }
-          .compact.last
+          .for_worker(@worker).last[:last_time]
       end
 
       def total_runtime
-        values.inject(:+) || 0.0
+        values(:total_time).inject(:+) || 0.0
       end
 
       def average_runtime
-        count = values.count
+        averages = values(:average_time)
+        count = averages.count
         return 0.0 if count == 0
-        total_runtime / count
+        averages.inject(:+) / count
```

Comment:
All of these calculations can be done based on the timings list I suggested above.